airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/configuration.py

```diff
@@ -46,11 +52,10 @@ class FormatConfiguration:
         format_license_container,
         [f"addlicense -c 'Airbyte, Inc.' -l apache -v -f {LICENSE_FILE_NAME} ."],
     ),
-    # Run isort and black on all python files.
     FormatConfiguration(
         Formatter.PYTHON,
         ["**/*.py"],
         format_python_container,
-        ["poetry run isort --settings-file pyproject.toml .", "poetry run black --config pyproject.toml ."],
+        ["poetry run ruff format ."],
```

Comment:
Please note that nested `pyproject.toml` file will not be considered.
I don't know if `ruff` traverse the whole repo and parses the nested `pyproject.toml` (I understood it was the opposite). In any case:
- `.` refers to the repo root
- nested `pyproject.toml` are not mounted to the formatting container. You'd have to modify the file filter [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/configuration.py#L52) if you want to mount them

Reply:
> I think ruff format is not running isort. ruff check is.

Maybe, but that part I'm more or less fine with. Doing full `ruff check` is not yet realistic for our repository overall, most packages don't quite comply with it just yet, we will need to make two passes. One for formatting, the second one for linting. 

> I also think the ruff commands should be called with --fix so that when you run airbyte-ci format fix all you actually get the fixed code exported back to the host.

Good catch — I completely phased out that there is `airbyte-ci format check` and `format fix`, and I need to make sure I'm doing the right thing there. If I change them, I need to change both. Will work on that. 

> Please note that nested pyproject.toml file will not be considered.

I need to experiment and verify this. My gut feeling is that you might be right on this. My HOPE is that when we run `ruff format .`, it walks the directory tree and applies the closest parent `pyproject.toml` config for each file in the tree. I.e. it will apply special rules for `airbyte-ci` but generic rules for connectors. But it MIGHT be that I'm wrong, and that it picks only one config file per invocation of ruff, i.e. everything will get global config.